src/providers/ipa/ipa_deskprofile_rules_util.c

```diff
@@ -0,0 +1,796 @@
+                  "directory \"%s\" that would be used to store Desktop "
+                  "Profiles rules users' directory[%d]: %s\n",
+                  domain_dir, ret, sss_strerror(ret));
+            goto done;
+        }
+    }
+
+    if (!S_ISDIR(info.st_mode)) {
+        ret = ENOTDIR;
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "\"%s\" exists but is not a directory. This should never happen!",
+              domain_dir);
+        goto done;
+    }
+
+    ret = EOK;
+
+done:
+    return ret;
+}
+
+errno_t
+ipa_deskprofile_rules_create_user_dir(const char *user_dir,
+                                      const char *domain_dir,
+                                      uid_t uid,
+                                      gid_t gid)
+{
+    struct stat info;
+    errno_t ret;
+
+    ret = ipa_deskprofile_rules_create_domain_dir(domain_dir);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_TRACE_FUNC,
+              "Failed to create the directory \"%s\" that would be used to "
+              "store the Desktop Profile rules users' directory [%d]: %s\n",
+              domain_dir, ret, sss_strerror(ret));
+        goto done;
+    }
+
+    ret = stat(user_dir, &info);
+    if (ret != EOK) {
+        ret = mkdir(user_dir, 0600);
+        if (ret != EOK) {
+            ret = errno;
+            DEBUG(SSSDBG_TRACE_FUNC,
+                  "Failed to create the directory \"%s\" that would be used "
+                  "to store the Desktop Profile rules [%d]: %s\n",
+                  user_dir, ret, sss_strerror(ret));
+            goto done;
+        }
+
+        /* As the directory is newly created, get an updated info about it */
+        ret = stat(user_dir, &info);
+        if (ret != EOK) {
+            ret = errno;
+            DEBUG(SSSDBG_TRACE_FUNC,
+                  "Failed to get updated information about the newly created "
+                  "directory \"%s\" that would be used to store Desktop "
+                  "Profiles rules [%d]: %s\n",
+                  user_dir, ret, sss_strerror(ret));
+            goto done;
+        }
+    }
+
+    if (!S_ISDIR(info.st_mode)) {
+        ret = ENOTDIR;
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "\"%s\" exists but is not a directory. This should never happen!",
+              user_dir);
+        goto done;
+    }
+
+    ret = chown(user_dir, uid, gid);
+    if (ret != EOK) {
+        ret = errno;
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "Failed to own the user directory \"%s\" [%d]: %s\n",
+              user_dir, ret, sss_strerror(ret));
+        goto done;
+    }
+
+    ret = EOK;
+
+done:
+    return ret;
+}
+
+static errno_t
+ipa_deskprofile_get_normalized_rule_name(TALLOC_CTX *mem_ctx,
+                                         const char *name,
+                                         char **_rule_name)
+{
+    TALLOC_CTX *tmp_ctx;
+    char buffer[1024];
+    size_t buffer_len;
+    size_t name_len;
+
+    tmp_ctx = talloc_new(mem_ctx);
```

Comment:
It was a left-over, as tmp_ctx was not even being used in this function.

Fixed locally!